AWS S3: Flaky test

In #50 we discovered a flaky test that can be resolved by rerunning the application. To reproduce locally, you can run `go test -count=10  ./...` and it _should_ fail with the following log:

```
2023/12/12 07:51:43 Exporting metrics for 1 regions
--- FAIL: TestCollector_Collect (0.00s)
    --- FAIL: TestCollector_Collect/cost_and_usage_output_-_result_with_nil_unit (0.00s)
        s3_test.go:630: 
                Error Trace:    /home/poko/dev/grafana/cloudcost-exporter/pkg/aws/s3/s3_test.go:630
                Error:          Received unexpected error:
                                
                                
                                Diff:
                                --- metric output does not match expectation; want
                                +++ got:
                                @@ -8,3 +8,3 @@
                                 # TYPE aws_s3_operations_cost gauge
                                -aws_s3_operations_cost{class="StandardStorage",region="ap-northeast-1",tier="1"} 1000
                                +aws_s3_operations_cost{class="StandardStorage",region="ap-northeast-1",tier="1"} +Inf
                                 
                Test:           TestCollector_Collect/cost_and_usage_output_-_result_with_nil_unit
        CostExplorer.go:108: PASS:      GetCostAndUsage(string,string,string)
```

--- a/pkg/aws/s3/s3.go
+++ b/pkg/aws/s3/s3.go
@@ -394,10 +394,16 @@ func exportMetrics(s3BillingData S3BillingData, m Metrics) {
 // unitCostForComponent will calculate the unit cost for a given component. This is necessary because the
 // unit cost will depend on the type of component.
 func unitCostForComponent(component string, pricing *Pricing) float64 {
+	// If the usage is 0, we don't want to divide by 0 which would result in NaN metrics _or_ +Inf
+	// TODO: Assess if we should return the pricing.Cost instead
+	if pricing.Usage == 0 {
+		log.Printf("Usage is 0 for component: %s\n", component)
+		return 0
+	}
+
 	switch component {
 	case "Requests-Tier1", "Requests-Tier2":
-		unitCost := pricing.Cost / (pricing.Usage / 1000)
-		return unitCost
+		return pricing.Cost / (pricing.Usage / 1000)
 	case "TimedStorage":
 		return (pricing.Cost / HoursInMonth) / pricing.Usage
 	default:

